### PR TITLE
Move setpoint calculation from zone time step to system time step

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
This is a followup issue for OnOff Thermostat. When the condition is satisfied, the setpoint is revised at every system time step, instead of every zone time step. 

Differences are expected in SingleFamilyHouse_TwoSpeed_CutoutTemperature.idf

There are other 5 files with differences due to report frequency change from "Zone" to "System": ZoneThermostatSetPointLo and ZoneThermostatSetPointHi.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by lgu1234** [comment]
@mjwitte I uploaded the code a while ago for review.

**Comment by Myoldmopar** [comment]
@lgu1234 two test files are failing to run, not just showing diffs.  It is an array index problem, so be on the lookout.  Since we are right against the freeze date, I'm going to anticipate we won't get this addressed and re-reviewed, so I'm going to put this in the next milestone.  If we magically get done with it that's great too.

**Comment by lgu1234** [comment]
@Myoldmopar I am going to take a look and hope I will fix today.

**Comment by lgu1234** [comment]
@Myoldmopar I fixed failure and merge the code again. I need to get this new feature for the spring release. I am going to change milestone.

File: src/EnergyPlus/ZoneTempPredictorCorrector.cc [hunk 1]
```diff
@@ -3340,7 +3331,7 @@ namespace ZoneTempPredictorCorrector {
 				LoadToHeatingSetPoint = AIRRAT( ZoneNum ) * ( TempZoneThermostatSetPoint( ZoneNum ) - ZoneT1( ZoneNum ) ) + TempDepZnLd( ZoneNum ) * ( TempZoneThermostatSetPoint( ZoneNum ) ) - TempIndZnLd( ZoneNum );
 			}
 			if ( RAFNFrac > 0.0 ) LoadToHeatingSetPoint = LoadToHeatingSetPoint / RAFNFrac;
-			if ( TempControlledZone( RelativeZoneNum ).HeatOffFlag ) {
+			if ( HeatOffFlag ) {
 				LoadToHeatingSetPoint = 0.0;
```
Comment (mjwitte) [inline src/EnergyPlus/ZoneTempPredictorCorrector.cc]: @lgu1234 I don't understand why `LoadToHeatingSetPoint` is set to 0.0 here. I think I follow the `HeatOffFlag` logic, but once the zone reaches the setpoint+cutoff and `HeatOffFlag=true`, then the thermostat setpoint should revert to the lower temperature and the LoadToHeatSetPoint should be based on that (the original setpoint).
@Myoldmopar This behaves far better than what's in develop, but I think there's one more improvement that could be made. I can experiment with that tomorrow or we merge this as-is and add the improvement post-release if it proves to be useful.

Comment (Myoldmopar) [reply src/EnergyPlus/ZoneTempPredictorCorrector.cc]: I will move this to the end of my list of tabs as they are ordered by when they should drop in.  If you have some time to experiment tomorrow that would be fine.  If it doesn't make it into this, that's fine too.

Comment (lgu1234) [reply src/EnergyPlus/ZoneTempPredictorCorrector.cc]: @mjwitte Thanks for your comments. I think of the same way by removing line LoadToHeatingSetPoint = 0.0. I will test tonight and let you know what happens.

**Comment by lgu1234** [comment]
@mjwitte I removed reset of load prediction. The results are better. The indoor temperature is between 22 and 24 during heating and 24.6 and 26.6 during cooling. Please let me know your comments after your experiment.

**Comment by Myoldmopar** [comment]
Thanks @lgu1234; I just merged in the latest develop branch so no unrelated diffs will pop up.  I'm letting CI run overnight on these branches before merging anything else, so the CI results will only be related to the changes on this branch.  If @mjwitte agrees with these changes and CI is happy it can go in in the morning.

**Comment by lgu1234** [comment]
@Myoldmopar Thanks.

**Pull request merged**